src/Chapter2.hs

```diff
@@ -858,7 +894,9 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [Int] -> [Int]
+rewind [] = []
+rewind xs = last xs : rewind (init xs)
```

Comment:
Your solution is correct! However, it is slow. In lists, it is quite slow to add or take anything at the end of the list. That is why it is always better to rewrite it with the `:` cons. Remember the explanation with trains? 🚂 🚋 🚋 But `last` function is very slow and need to walk the whole list at each iteration.

That is why a more efficient solution is with the accumulator and the recursive function that will do the addition at the start of the list which is instant!

You can read a bit more about the `go` pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go